server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -1390,7 +1366,11 @@ protected int comparePivot(int j) {
         }
 
         public float delta() {
-            return weights[weights.length - 1] - weights[0];
+            return weights.length == 0 ? 0.0f : weights[weights.length - 1] - weights[0];
```

Comment:
It used to be impossible to have zero nodes in a sorter, but now that can happen in the search partition